hackage-security/hackage-security.cabal

```
@@ -58,8 +58,8 @@ flag Cabal-syntax
 
 flag lukko
   description: Use @lukko@ for file-locking, otherwise use @GHC.IO.Handle.Lock@
-  manual:      True
-  default:     True
+  manual:      False
```

```suggestion
  manual:      True
```
There is no point to make this flag automatic and false by default - Cabal would never have an incentive to flip it.

---

Wouldn't it flip the flag when using old enough GHCs?

---

Why? The package unconditionally mandates `base >= 4.11`, the flag does not affect this.

---

You're right. I saw [this line](https://github.com/haskell/hackage-security/blob/d160ca4e6ec562dab0bd226a0b19e6436091179c/hackage-security/hackage-security.cabal#L137) and assumed it wasn't just there for better error messages so I didn't notice [this other line](https://github.com/haskell/hackage-security/blob/d160ca4e6ec562dab0bd226a0b19e6436091179c/hackage-security/hackage-security.cabal#L110).

---

It used to be, but since then the package dropped compatibility with old GHCs for unrelated reasons.